control-plane-operator/controllers/hostedcontrolplane/v2/etcd/component.go

```diff
@@ -0,0 +1,69 @@
+package etcd
+
+import (
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	component "github.com/openshift/hypershift/support/controlplane-component"
+)
+
+const (
+	ComponentName = "etcd"
+)
+
+var _ component.ComponentOptions = &etcd{}
+
+type etcd struct {
+}
+
+// IsRequestServing implements controlplanecomponent.ComponentOptions.
+func (e *etcd) IsRequestServing() bool {
+	return false
+}
+
+// MultiZoneSpread implements controlplanecomponent.ComponentOptions.
+func (e *etcd) MultiZoneSpread() bool {
+	return true
+}
+
+// NeedsManagementKASAccess implements controlplanecomponent.ComponentOptions.
```

Comment:
It'd be nice if we document some of these as we go through.
E.g. // etcd needs management kas access for some controllers to implement leader election, e.g. etcd-defrag-controller

Reply:
done